### PR TITLE
DB optimizations

### PR DESCRIPTION
**Pull request by mic4ael**
Screenshots of the expensive requests

before
screenshot from 2018-07-06 13-48-13

and after
screenshot from 2018-07-06 13-47-22

Unfortunately, to avoid passing a big list of room ids to the RH I query again for the available list of rooms which makes the total time of SQL queries a bit higher.

### CODE REVIEW

File: indico/modules/rb_new/util.py [hunk 1]
```diff
@@ -154,34 +160,48 @@ def get_existing_room_occurrences(room, start_dt, end_dt, allow_overlapping=Fals
...
 
-def get_room_conflicts(room, start_dt, end_dt, repeat_frequency, repeat_interval):
-    conflicts = []
-    pre_conflicts = []
 
+def get_rooms_conflicts(rooms, start_dt, end_dt, repeat_frequency, repeat_interval):
+    room_conflicts = {}
     candidates = ReservationOccurrence.create_series(start_dt, end_dt, (repeat_frequency, repeat_interval))
-    occurrences = ReservationOccurrence.find_overlapping_with(room, candidates).all()
+    room_ids = map(attrgetter('id'), rooms)
```
Comment (ThiefMaster) [inline indico/modules/rb_new/util.py]: room_ids = [r.id for r in rooms]

File: indico/modules/rb_new/util.py [hunk 2]
```diff
@@ -154,34 +160,48 @@ def get_existing_room_occurrences(room, start_dt, end_dt, allow_overlapping=Fals
...
+    query = (ReservationOccurrence
+             .find(Reservation.room_id.in_(room_ids),
```
Comment (ThiefMaster) [inline indico/modules/rb_new/util.py]: we don't use `.find()` in new code; use `.query` and `.filter()` instead

Comment (pferreir) [reply indico/modules/rb_new/util.py]: :confused: Really?

File: indico/modules/rb_new/util.py [hunk 3]
```diff
@@ -141,8 +141,13 @@ def get_buildings():
 
 
 def get_existing_room_occurrences(room, start_dt, end_dt, allow_overlapping=False, only_accepted=False):
+    return get_existing_rooms_occurrences([room], start_dt, end_dt, allow_overlapping, only_accepted).get(room.id, [])
+
+
```
Comment (mvidalgarcia) [inline indico/modules/rb_new/util.py]: Extra empty line

Comment (mic4ael) [reply indico/modules/rb_new/util.py]: where? there are only two lines which are standard

File: indico/modules/rb_new/client/js/components/pages/BookRoom.jsx [hunk 4]
```diff
@@ -33,6 +33,8 @@ import Room from '../../containers/Room';
 import Timeline from '../../containers/Timeline';
 import BookRoomModal from '../../containers/BookRoomModal';
 import roomDetailsModalFactory from '../modals/RoomDetailsModal';
+import {roomPreloader} from '../../util';
+
```
Comment (meluru) [inline indico/modules/rb_new/client/js/components/pages/BookRoom.jsx]: I guess this empty line is not needed :)

File: indico/modules/rb_new/controllers/backend.py [hunk 5]
```diff
@@ -42,6 +42,8 @@
 from indico.web.util import jsonify_data
 
 
+NUMBER_OF_SUGGESTIONS = 5
```
Comment (ThiefMaster) [inline indico/modules/rb_new/controllers/backend.py]: `NUM_SUGGESTIONS`? "of" is one of these words that usually make variable names overly verbose

File: indico/modules/rb_new/controllers/backend.py [hunk 6]
```diff
@@ -127,17 +129,22 @@ def _process(self):
...
     def _process(self, args):
-        rooms = Room.query.filter(Room.is_active, Room.id.in_(args.pop('room_ids')))
-        date_range, availability = get_rooms_availability(rooms, **args)
+        query = search_for_rooms(args, only_available=True)
+        if 'limit' in args:
+            query = query.limit(args.pop('limit'))
+
+        rooms = query.all()
+        if 'additional_room_ids' in args:
+            rooms.extend(Room.query.filter(Room.id.in_(args.pop('additional_room_ids'))))
```
Comment (ThiefMaster) [inline indico/modules/rb_new/controllers/backend.py]: I see a query that does not exclude deleted rooms

**Pull request merged**